src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -128,32 +132,71 @@ public async Task<(Orchestrator, Recommendation, CloudApplication)> InitializeDe
                     _awsResourceQueryer,
                     _deploymentBundleHandler,
                     _dockerEngine,
-                    new[] { RecipeLocator.FindRecipeDefinitionsPath() });
+                    _customRecipeLocator,
+                    new List<string> { RecipeLocator.FindRecipeDefinitionsPath() });
 
             // Determine what recommendations are possible for the project.
-            var recommendations = await GenerateDeploymentRecommendations(orchestrator);
+            var recommendations = await GenerateDeploymentRecommendations(orchestrator, deploymentProjectPath);
+
+            // Get all existing applications that were previously deployed using our deploy tool.
+            var allDeployedApplications = await _deployedApplicationQueryer.GetExistingDeployedApplications();
 
-            // Look to see if there are any existing deployed applications using any of the compatible recommendations.
-            var deployedApplications = await _deployedApplicationQueryer.GetExistingDeployedApplications(recommendations);
+            // Filter compatible applications that can be re-deployed  using the current set of recommendations.
+            var compatibleApplications = GetCompatibleApplications(allDeployedApplications, recommendations);
 
             // Get Cloudformation stack name.
-            var cloudApplicationName = GetCloudApplicationName(stackName, userDeploymentSettings, deployedApplications);
+            var cloudApplicationName = GetCloudApplicationName(stackName, userDeploymentSettings, compatibleApplications);
 
             // Find existing application with the same CloudFormation stack name.
-            var deployedApplication = deployedApplications.FirstOrDefault(x => string.Equals(x.Name, cloudApplicationName));
+            var deployedApplication = allDeployedApplications.FirstOrDefault(x => string.Equals(x.Name, cloudApplicationName));
 
             Recommendation? selectedRecommendation = null;
             if (deployedApplication != null)
+            {
+                // Verify that the target application can be deployed using the current set of recommendations
+                if (!compatibleApplications.Any(app => app.StackName.Equals(deployedApplication.StackName, StringComparison.Ordinal)))
+                {
+                    var errorMessage = $"{deployedApplication.StackName} already exists as a Cloudformation stack but a compatible recipe to perform a redeployment was no found";

```

Comment:
Recipe is a term used for the objects used in our implementation. From the user's point of view unless they are advanced writing their own recipe it is just an implementation detail that we should be talking to them like they know how we implemented the tool. What they see is a list of recommendations and that is how we should talk to them as far as deployment goes.

Suggested text:

"$"{deployedApplication.StackName} already exists as a Cloudformation stack but it was created with an incompatible recommendation for this application."